src/shinylive.ts

```diff
@@ -123,11 +123,19 @@ async function createAndOpenShinyliveLink(
  * files will be saved. The link is decoded and the files are saved into the
  * directory.
  *
+ * @param {string} [url] The Shinylive URL to save the app from. If not provided
+ * the user will be prompted to enter a URL.
+ *
  * @export
  * @async
  */
-export async function shinyliveSaveAppFromUrl(): Promise<void> {
-  const url = await askUserForUrl();
+export async function shinyliveSaveAppFromUrl(
```

Comment:
No need to change this for this PR, but I found the way single file vs. multiple file shinylive projects are handled, kind of confusing; there are several points in this function and in `askUserForOutputLocation()` where the logic varies depending on whether you're doing single vs. multiple that is a bit hard for me to reason about. I suspect it would make more sense to me if e.g. there were separate `askUserForOutputDir` and `askUserForOutputFile` functions, and a single conditional in this function with slightly heftier branches.

Reply:
Yeah that makes a lot of sense. I definitely have a similar feeling coming back to that code. I'll refactor this in a separate PR.